tools/horizon-cmp/internal/response.go

```
@@ -0,0 +1,134 @@
+}
+
+func NewResponse(domain, path string) *Response {
+	response := &Response{
+		Domain: domain,
+		Path:   path,
+	}
+
+	resp, err := http.Get(domain + path)
+	if err != nil {
+		panic(err)
+	}
+
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusNotFound {
+		panic(resp.StatusCode)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		panic(err)
+	}
+
+	response.Body = string(body)
+
+	normalizedBody := response.Body
+	// `result_meta_xdr` can differ between core instances (confirmed this with core team)
+	normalizedBody = findResultMetaXDR.ReplaceAllString(normalizedBody, "")
+	// Remove Horizon URL from the _links
+	normalizedBody = strings.Replace(normalizedBody, domain, "", -1)
+
+	// normalizedBody = succ1.ReplaceAllString(normalizedBody, "")
+	// normalizedBody = succ2.ReplaceAllString(normalizedBody, "")
+	// normalizedBody = succ3.ReplaceAllString(normalizedBody, "")
+	// normalizedBody = succ4.ReplaceAllString(normalizedBody, "")
+	// normalizedBody = succ5.ReplaceAllString(normalizedBody, "")
+
+	response.NormalizedBody = normalizedBody
+	return response
+}
+
+func (r *Response) Equal(other *Response) bool {
+	return r.NormalizedBody == other.NormalizedBody
+}
+
+func (r *Response) SaveDiff(outputDir string, other *Response) {
+	if r.Path != other.Path {
+		panic("Paths are different")
+	}
+
+	fileName := pathToFileName(r.Path)
+
+	if len(fileName) > 100 {
+		fileName = fileName[0:100]
+	}
+
+	fileA := fmt.Sprintf("%s/%s.old", outputDir, fileName)
+	fileB := fmt.Sprintf("%s/%s.new", outputDir, fileName)
+	fileDiff := fmt.Sprintf("%s/%s.diff", outputDir, fileName)
+
+	err := ioutil.WriteFile(fileA, []byte(r.Path+"\n\n"+r.Body), 0744)
+	if err != nil {
+		panic(err)
+	}
+
+	err = ioutil.WriteFile(fileB, []byte(other.Path+"\n\n"+other.Body), 0744)
+	if err != nil {
+		panic(err)
+	}
+
+	out, err := exec.Command("diff", fileA, fileB).Output()
+	if err != nil {
+		// Ignore, user will generate diff manually
```

Why do we want to ignore the error returned by the `Output` given that the user will generate `diff` program manually? And if it makes more sense to ignore the error, maybe we can just do:
```
out, _ := exec.Command("diff", fileA, fileB).Output()
```

---

Done.